docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -1238,6 +1238,130 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
+                configRootName, // name of the config root pertaining to this extension
+                new Tag("datasource", dataSource.getName())));
+    }
+}
+----
+
+In this example, there is a gauge per each data source that shows the current number of active connections.
+In this example, we provide an object that implements the logic of the metric, in the case of a gauge,
+it needs to implement `org.eclipse.microprofile.metrics.Gauge`. Example:
+
+[source%nowrap.java]
+----
+public class ActiveCountGauge implements Gauge<Long> {
+
+    private String dataSourceName;
+
+    public ActiveCountGauge() {
+
+    }
+
+    public ActiveCountGauge(String dataSourceName) {
+        this.dataSourceName = dataSourceName;
+    }
+
+    @Override
+    public Long getValue() {
+        DataSource ds = obtainDataSource(dataSourceName); // some logic to get hold of the data source
+        return ds.getMetrics().getActiveCount();
+    }
+}
+----
+
+When registering gauges, it is required to provide a custom object that implements the correct metric type.
+For metric types other than gauges, this is optional - you can choose whether to provide an implementation,
+or whether the default implementation class from `SmallRye Metrics` should be instantiated for the metric and registered
+in the metric registry. To update the metric value, it will be necessary to look them up in the metric registry
+and call the methods specific to each metric type.
+
+Look into the `AgroalProcessor#registerMetrics` method for an example how this was done for the Agroal extension.
+
+==== Case 3: It is necessary to collect metrics within the extension code
+
+In this case, the dependency setup and build-time registration is the same as in case 2. The difference will be that instead of introducing
+custom metric objects that bridge between library-specific metrics and MP Metrics, the extension's code will contain metric collection code like this:
+
+[source%nowrap.java]
+----
+public void methodThatShouldBeCounted() {
+    if(metricsEnabled) {
+        Counter counter = MetricRegistries.get(MetricRegistry.Type.VENDOR).counter("name");
+        counter.inc();
+    }
+    // proceed with the invocation normally
+}
+----
+
+In this case, be aware that the MP Metrics API and implementation might not be available in the runtime classpath!
+This is why there is a `if(metricsEnabled)` check that should make sure that if metrics are not enabled, we don't touch MP Metrics classes,
+because that would fail. The extension itself will need to provide such check, and it must return true only if both of these conditions hold:
+
+- The Metrics capability is present and therefore the SmallRye Metrics library is available
+- Metrics are not disabled for this extension
+
+A recommended way to implement this check would be to produce a specific `SystemPropertyBuildItem` during build, where it is easy
+to evaluate these two conditions, and then check back to this system property at runtime.
```

Comment:
Perhaps show an implementation of `metricsEnabled`?

Reply:
I added an example how to produce the system property